spyder/app/tour.py

```diff
@@ -982,13 +1044,22 @@ def _process_widgets(self, names, spy_window):
         dockwidgets = []
 
         for name in names:
-            base = name.split('.')[0]
-            temp = getattr(spy_window, base)
-
-            # Check if it is the current editor
-            if 'get_current_editor()' in name:
-                temp = temp.get_current_editor()
-                temp = getattr(temp, name.split('.')[-1])
+            try:
+                base = name.split('.')[0]
+                try:
+                    temp = getattr(spy_window, base)
+                except AttributeError:
+                    for item in spy_window.thirdparty_plugins:
+                        if type(item).__name__ == name:
+                            temp = item
+                            break

```

Comment:
```suggestion
                            break
                    else:
                        temp = None
```

We need this so that the `AttributeError` passes through to the outer except block if the item isn't found in this list.